include/SDL3/SDL_video.h

```diff
@@ -1425,6 +1425,31 @@ extern DECLSPEC int SDLCALL SDL_RestoreWindow(SDL_Window *window);
  */
 extern DECLSPEC int SDLCALL SDL_SetWindowFullscreen(SDL_Window *window, SDL_bool fullscreen);
 
+/**
+ * Block until any pending window state is finalized.
+ *
+ * On asynchronous windowing systems, this acts as a synchronization barrier for
+ * pending window state. It will attempt to wait until any pending window state
+ * has been applied and is guaranteed to return within finite time. Note that for
+ * how long it can potentially block depends on the underlying window system, as
+ * window state changes may involve somewhat lengthy animations that must complete
+ * before the window is in its final requested state.
+ *
+ * On windowing systems where changes are immediate, this does nothing.
+ *
+ * \param window the window for which to wait for the pending state to be applied
+ *
+ * \since This function is available since SDL 3.0.0.
+ *
+ * \sa SDL_SetWindowSize
+ * \sa SDL_SetWindowPosition
+ * \sa SDL_SetWindowFullscreen
+ * \sa SDL_MinimizeWindow
+ * \sa SDL_MaximizeWindow
+ * \sa SDL_RestoreWindow
+ */
+extern DECLSPEC void SDLCALL SDL_SyncWindow(SDL_Window *window);
```

Comment:
Is this function infallible?

Reply:
Outside of an invalid window pointer, it can't fail, but the backend call can time out while waiting. The application can't do anything to fix or recover from this, but it could keep waiting if desired. It probably should return an int to communicate exactly what's happening:
- result < 0 = error (bad parameter)
- 0 = success
- result > 0 = timed out (maybe the app wants to try and wait again?)